src/plugin/helpers.cc

```diff
@@ -57,14 +57,23 @@ namespace chapel {
     return true;
   }
 
+  string GetPackageName(const FileDescriptor* descriptor) {
+    return UnderscoresToCamelCase(descriptor->package());
```

Comment:
I am not so sure about removing underscores in the name. Is this something we've talked about before? I have thought it would just have the package name - provided there are no illegal characters in the filename.

Reply:
I'm curious if protobuf compilers in other languages do this sort of thing.

Reply:
> I am not so sure about removing underscores in the name. Is this something we've talked about before?

No, I don't think so. While removing underscores I had in mind the general camel case convention followed in Chapel. I generally take `_` to be used for separating words, e.g - my_package and in Chapel we use camel case for that(myPackage), so I thought it might be useful to have underscores get converted.  

> provided there are no illegal characters in the filename.

Now, I am taking only alphanumeric chars to be valid chars. What other characters can we consider as valid(or some that we can consider invalid?)?

Reply:
`-` is a character that could be in a filename that does not work as a module name, off the top of my head

Reply:
And in our case we want the filename and the module name to be same. :confused:

Reply:
> I think we want it to be as reasonably close as possible, and to clearly define the behavior when it would differ.

But AFAIK the module name and the file name should be the same? Aren't they?

Reply:
> But AFAIK the module name and the file name should be the same? Aren't they?

Hmm, maybe this is something that we should discuss more.  I don't know that I have a strong opinion on which way is correct, what's most important to me is that we are consistent with other protobuf compilers and that the behavior is clearly documented so that users know what to expect when they provide a .proto file whose name would not be a valid module name in Chapel.

Oh, and it looks like underscore is the only non alphanumeric character allowed in names, if the [spec](https://chapel-lang.org/docs/latest/language/spec/lexical-structure.html#identifiers) is up to date with the compiler.

Reply:
This is how Protobuf Style guide suggests writing package names (https://developers.google.com/protocol-buffers/docs/style#packages)

> Package name should be in lowercase, and should correspond to the directory hierarchy. e.g., if a file is in my/package/, then the package name should be my.package.

Reply:
I tried playing around with some packages names in a proto file and I observe that the `protoc` compiler only allows `_`, `.` and alphanumeric characters in package name and throws for other characters. For example in C++ ->

```
package my_package;
```
gives 

```
namespace my_package {
```
and 

```
package my.package;
```
gives

```
namespace my {
namespace package {
```

The generated file names are always derived from the `proto` file name in other languages.

If Chapel only allows `_` we can keep them intact both for module name and the file name. But if user do not specify `package` then there is a need to endure that the module name do not contain restricted characters. We can to include this in the user guide, maybe under a "Naming of proto files"?